### ISSUE TITLE
Fighting voter apathy

### ISSUE THREAD

**Issue by sschiessl-bcp** [bitshares/bitshares-ui#1934]
Voting should be highlighted in the UI. This issue is means to create a component that shows prominently
 - upcoming worker proposals
 - highlighting new proposals

Furthermore, maybe via a expandable list
 - currently active worker proposals

User should be able to mark proposals as "seen" (setting stored locally), which removes them from the new proposals highlighted list.

**Comment by grctest** [comment]
Beat you to it by about an hour ;D https://github.com/bitshares/bitshares-ui/issues/1932

**Comment by sschiessl-bcp** [comment]
Your issue is post BSIP 22

**Comment by grctest** [comment]
Ah true, however it's not explicitly delaying related dev activity/discussion to post BSIP 22 implementation, just highlighting that it will amplify the consequences of voter apathy if we don't take action. Voter apathy is a general BTS problem, not solely 22 related, you're right there.

Related: https://github.com/bitshares/bitshares-ui/issues/1339

**Comment by kristinlow** [comment]
Zooming out a little beyond the specific components to the behavioural outcome, I think we would see apathy recede if: 

- People are firstly aware that they can vote on bitshares (this implies they know how voting works, what workers are, what BSIPs are, etc) 
- That people feel encouraged, if not obligated, to participate in voting
- That people feel confident of their votes' importance/significance, no matter the size of their BTS holdings

Ideas (some of which mentioned by others previously) 

- Add voting tab to main menu at top
- Complete overhaul of worker section on current UI, focusing on the content or decision being asked, highlighting the creator, visually distinguishing between BSIP and non-BSIP workers, visually enhancing the display of active, retired, proposed workers, etc) 
- Maybe highlight the ratio of BTS that someone can vote with under their account name in the top menu, to prompt them that they can "spend" their votes. 

Good to pin a measurement to this. Is there an easy way to track total voter participation currently? We could use this to benchmark any changes we make. 

Please add more ideas, I would like to mockup what some of these could look like. 

PS: This is my first contribution to any bitshares stuff, hoping to be more active and useful in future. Forgive any errors in the above.

**Comment by Bluffet** [comment]
Bringing it to people's attention certainly will help.

Another thing is making the proposals visible within client, possible standardized and stored for reference and to easily compare them. It's nice to see this being put forward again.

**Comment by abitmore** [comment]
Here is an old thread related in forum: https://bitsharestalk.org/index.php/topic,20962.0.html

**Comment by gibbsfromncis** [comment]
Definitely requires UX.

I’ve assigned Alex to this.

**Comment by wmbutler** [comment]
I'm not sure I see this as a priority. We need to fix the problem of WP's being around forever (those where the start date has passed and they have an expiration date of 2020 and never received enough votes before we can present a proper current list of items to vote on.

**Comment by gibbsfromncis** [comment]
@wmbutler I like the idea to highlight unvisible features for common user. But you're right it is not a priority right now. 

I prefer to push our UX resources to another important parts

**Comment by sschiessl-bcp** [comment]
> We need to fix the problem of WP's being around forever (those where the start date has passed and they have an expiration date of 2020 and never received enough votes before we can present a proper current list of items to vote on.

This is a mere default setting away. Add a checkbox "Hide legacy proposals" that is enabled per default.
Issue #1986 adds it to current voting tabs.

> I'm not sure I see this as a priority. 

Voter apathy is one of the most dominant topics dicussed in the channels, certainly a priority from the community's point of view

**Comment by abitmore** [comment]
Just came up with an idea: we can add a "last vote time" field in core. Base on it, we can show a notification on the top of UI, which says: it's been xxx days since last time you updated your votes, click here for how voting works, click here for what/who to vote. UX is needed.

Ideally, to reduce disturbances, the notification should only show when the user has >= X BTS in his account which may be distributed in these places:
* balance
* collateral or market orders (can get from `core_in_orders` field in `account_statistics_object`)
* cashback object

**Comment by clockworkgr** [comment]
> Just came up with an idea: we can add a "last vote time" field in core. Base on it, we can show a notification on the top of UI, which says: it's been xxx days since last time you updated your votes, click here for how voting works, click here for what/who to vote. UX is needed.
> 
> Ideally, to reduce disturbances, the notification should only show when the user has >= X BTS in his account which may be distributed in these places:
> 
> * balance
> * collateral or market orders (can get from `core_in_orders` field in `account_statistics_object`)
> * cashback object

hmmmm

cashback object is counted towards voting weight? just that one or all vesting balances?

was under the impression vesting balances didnt count

**Comment by sschiessl-bcp** [comment]
> Just came up with an idea: we can add a "last vote time" field in core

I was aiming for a similar approach, but to hold something like the "last vote time" in localStorage for the UI. Your approach would certainly be more robust as its independent of the used UI.

**Comment by abitmore** [comment]
>cashback object is counted towards voting weight? just that one or all vesting balances?

Only cashback according to bitshares-core [code](https://github.com/bitshares/bitshares-core/blob/2.0.180823/libraries/chain/db_maint.cpp#L1110-L1112):
```
uint64_t voting_stake = stats.total_core_in_orders.value
      + (stake_account.cashback_vb.valid() ? (*stake_account.cashback_vb)(d).balance.amount.value: 0)
      + stats.core_in_balance.value;
```

**Comment by grctest** [comment]
Where are we on this one? What's the consensus on how to fight voter apathy?

I've raised a new UX issue to fight voter apathy re worker proposals: https://github.com/bitshares/bitshares-ui/issues/3750
